### PR TITLE
Tree View

### PR DESCRIPTION
**Pull request by winkerVSbecks**
- Main block view
- Side panel with state and other info
- Search for component
- Minor clean up and comments

@igor-ka @bertrandk please review and merge.

### CODE REVIEW

File: app/ts/batarangle/frontend/batarangle.ts [hunk 1]
```diff
@@ -12,40 +13,47 @@ import {BackendMessagingService} from './channel/backend-messaging-service';
...
-    private componentDataStore: ComponentDataStore) {
+    private componentDataStore: ComponentDataStore,
+    private lifeCycle: LifeCycle
+  ) {
 
     Rx.Observable.interval(1000).take(1).subscribe(() => {
       this.userActions.getComponentData();
     });
 
     chrome.devtools.inspectedWindow.eval(
```
Comment (igor-ka) [inline app/ts/batarangle/frontend/batarangle.ts]: this block can be removed, its a left over of our script injection "experiments"

File: app/ts/batarangle/frontend/batarangle.ts [hunk 2]
```diff
@@ -12,40 +13,47 @@ import {BackendMessagingService} from './channel/backend-messaging-service';
...
+  private tree: any;
   constructor(
     private backendAction: BackendActions,
     private userActions: UserActions,
-    private componentDataStore: ComponentDataStore) {
+    private componentDataStore: ComponentDataStore,
+    private lifeCycle: LifeCycle
+  ) {
 
     Rx.Observable.interval(1000).take(1).subscribe(() => {
```
Comment (igor-ka) [inline app/ts/batarangle/frontend/batarangle.ts]: We need to try it w/o the 1000ms delay, it should probably work w/o it now

File: app/ts/batarangle/frontend/core/tree-view/tree-view-api.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import {List} from 'immutable';
```
Comment (igor-ka) [inline app/ts/batarangle/frontend/core/tree-view/tree-view-api.ts]: This file can go as well i believe, this is a left over of our "mock data" api

File: app/ts/devtools.ts [hunk 4]
```diff
@@ -1,17 +1,17 @@
-/** 
+/**
  * Create an Angular panel from the Component Tree View
  * on Chrome Development Tools window.
 */
 chrome.devtools.panels.create(
-  'Angular', 
-  'app/images/angular.png', 
-  'app/js/batarangle/frontend/batarangle.html', 
+  'Angular2',
```
Comment (igor-ka) [inline app/ts/devtools.ts]: I think we want to rename it to Batarangle

**Comment by igor-ka** [comment]
@winkerVSbecks it looks good varun, there are a couple of little things that can go that i mentioned in the comments other than that looks great to me.

File: app/ts/devtools.ts (hunk 4 above)
Comment (winkerVSbecks) [inline app/ts/devtools.ts]: There is `Batarangle` in the main panel. I think we should leave this one as Angular/Angular2. After all that's the framework we are debugging in this panel.

**Comment by winkerVSbecks** [comment]
@igor-ka updated to incorporate your feedback.

**Comment by igor-ka** [comment]
:shipit looks good.

**Pull request merged**